src/Discovery/DiscoveryDetails.tsx

```diff
@@ -282,6 +288,24 @@ const DiscoveryDetails = (props: Props) => {
               </Button>
             )
             : null}
+            {(
+              props.user.username
```

Comment:
I am not sure if this is the correct conditional logic.

Reply:
This is okay but the one in suggestion is more accurate. However, the line below is not right
```suggestion
              props.modalData[studyRegistrationConfig.studyRegistrationValidationField]
            && props.user.username
```